skimage/draw/_draw.pyx

```diff
@@ -524,6 +524,13 @@ def _ellipse_perimeter(Py_ssize_t r_o, Py_ssize_t c_o, Py_ssize_t r_radius,
         rr.extend(rr_t)
         cc.extend(cc_t)
 
+    # sorting on the basis of angle
+    rr = np.array(rr, dtype=np.intp)
+    cc = np.array(cc, dtype=np.intp)
+    angle = np.argsort(np.arctan2(cc-np.mean(cc),rr-np.mean(rr)))
+    rr = np.take(rr, angle, axis=0)
+    cc = np.take(cc, angle, axis=0)
+
```

Comment:
Hey @anamfatima1304, thanks for working on this! In case you are not aware, this fails because this file is written in Cython. Similar to C, the variables `rr` and `cc` were declared with `cdef list rr = ...` so you can't assign an array to them because it is a different type. 

Instead I would propose that you move this snippet to the Python level in [`ellipse_perimeter`](https://github.com/scikit-image/scikit-image/blob/54316d5f640675b92fa5dfb553f4cca9f23c9d40/skimage/draw/draw.py#L645-L713).